daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/unparsers/UState.scala

```diff
@@ -78,6 +79,24 @@ abstract class UState(
   extends ParseOrUnparseState(vbox, diagnosticsArg, dataProcArg, tunable)
   with Cursor[InfosetAccessor] with ThrowsSDE with SavesErrorsAndWarnings {
 
+  /**
+   * Push onto the dynamic TRD context stack, so long as the argument is a TermRuntimeData.
+   * Otherwise it is ignored.
```

Comment:
Just curious, what types of things aren't TermRuntimeData's that we are pushing on the stack that hsould be ignored?

Reply:
I will change to TRD only. There's one place that needs a downcast but that's better. 
We dont' have a trait/base that is TermUnparser, so the choiceBranchMap maps choiceBranchEvent to Unparser, and an Unparser might not be a term. E.g., unparser for statements aren't terms.
Now, it just so happens that choiceBranchMap only contains Unparsers for Terms, so downcasting there is better, or I could mess with the choiceBranchMap, but I'd rather not get into that.